core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java

```diff
@@ -61,29 +63,18 @@ public Waiter isLatestBuildPresent(String buildConfigName) {
 	 */
 	public Waiter isProjectClean() {
 		BooleanSupplier bs = () -> {
-			List<Boolean> cleanedResources = new ArrayList<>();
-			cleanedResources.add(openShift.templates().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.apps().deployments().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.batch().jobs().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.deploymentConfigs().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.apps().statefulSets().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.replicationControllers().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.buildConfigs().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.imageStreams().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.endpoints().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.services().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.builds().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.routes().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.pods().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.persistentVolumeClaims().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.autoscaling().horizontalPodAutoscalers().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.configMaps().withoutLabel(OpenShift.KEEP_LABEL).list().getItems().isEmpty());
-			cleanedResources.add(openShift.getUserSecrets().isEmpty());
-			cleanedResources.add(openShift.getUserServiceAccounts().isEmpty());
-			cleanedResources.add(openShift.getUserRoleBindings().isEmpty());
-			cleanedResources.add(openShift.getRoles().isEmpty());
-
-			return !cleanedResources.contains(false);
+			List<HasMetadata> removables = openShift.listRemovableResources();
+
+			// HACK - Handle issue with hanging -deploy orphan
+			if (removables.isEmpty()) {
```

Comment:
isProjectClean() waiter should not contain the cleanFinalizers() hack, only the waiter returned from OpenShift::clean() should have the hack.